binding.cc

```diff
@@ -390,8 +390,7 @@ namespace zmq {
 
   void
   Socket::UV_PollCallback(uv_poll_t* handle, int status, int events) {
-    assert(status == 0);
-
+    if (status != 0) NanThrowError(ExceptionFromError());
```

Comment:
i doubt we should error check there but if we want to, then maybe a Type error would be best? what do you think @kkoopa ?